File: c/problems/problem#2962.c
long long countSubarrays(int *nums, int numsSize, int k) {
  long long ans = 0;
  int maxNum = nums[0];
  for (int i = 1; i < numsSize; i++) {
    if (nums[i] > maxNum) {
      maxNum = nums[i];
    }
  }
  int left = 0, right = 0;
  int count = 0;
  while (right < numsSize) {
    int n = nums[right];
    if (n == maxNum) {
      count++;
    }
    while (count == k) {
      if (nums[left] == maxNum) {
        count--;
      }
      left++;
    }
    ans += left;
    right++;
  }
  return ans;
}

File: go/problem#2962.go
package main

func countSubarrays2962(nums []int, k int) int64 {
	maxNum := nums[0]
	for _, n := range nums {
		if n > maxNum {
			maxNum = n
		}
	}
	count := 0
	ans := int64(0)
	l := int64(0)
	for _, n := range nums {
		if n == maxNum {
			count++
		}
		for count == k {
			if nums[l] == maxNum {
				count--
			}
			l++
		}
		ans += l
	}
	return ans
}
